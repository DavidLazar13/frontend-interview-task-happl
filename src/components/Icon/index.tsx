import { ICONS } from '@/constants/icons'
import { Icons } from '@/types/icons'
import React from 'react'

type Props = {
  icon: Icons
  width?: number | string
  height?: number | string
}

const Icon = ({ icon, width = 20, height = 20 }: Props) => {
  const Component = ICONS[icon]
  return (
    <Component
      width={width}
      height={height}
      data-testid={`icon-${icon}`}
      strokeWidth={1.5}
    />
  )
}

export default Icon
