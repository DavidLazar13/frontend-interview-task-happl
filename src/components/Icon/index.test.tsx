import { render } from '@testing-library/react'
import Icon from '@/components/Icon'
import '@testing-library/jest-dom'

describe('Icon', () => {
  it('renders as expected', () => {
    const { container } = render(<Icon icon="logo" />)
    expect(container).toMatchSnapshot()
  })
})
