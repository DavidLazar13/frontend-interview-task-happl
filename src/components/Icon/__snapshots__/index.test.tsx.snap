// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders as expected 1`] = `
<div>
  <svgr-url
    data-testid="icon-logo"
    height="20"
    strokewidth="1.5"
    width="20"
  />
</div>
`;
